### PR TITLE
fix(replay): Don't emit an EventView from useReplaysFromIssue when no replayIds are in the fetched list

### LINKED ISSUE #62166: Replays tab in Issue fails to load "Could not parse value for `id`"
Repro:
https://dragonfruit-dev.sentry.io/issues/4680186801/replays/?project=4504179109068800&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=90d&stream_index=13

The request that fails with 400: https://dragonfruit-dev.sentry.io/api/0/organizations/dragonfruit-dev/replays/?field=activity&field=browser&field=count_dead_clicks&field=count_errors&field=count_rage_clicks&field=duration&field=finished_at&field=id&field=is_archived&field=os&field=project_id&field=started_at&field=user&per_page=50&project=-1&query=id%3A%5B%5D&queryReferrer=issueReplays&sort=-started_at&statsPeriod=14d

Note: `&query=id%3A%5B%5D`

### PR DESCRIPTION
**Pull request by ryan953**
When we fetch the `/replay-count/` (list of replayIds) inside of `useReplaysFromIssue`, we need to guard against the empty list being returned. If the list is empty then there are no replayIds and we can skip the next fetch altogether.

Also I updated all the `queryReferrer` prop values to make stuff like this easier to track down.

SCR-20231220-oein

Fixes https://github.com/getsentry/sentry/issues/62166

### CODE REVIEW

**Review by michellewzhang — APPROVED** [review]

File: static/app/views/issueDetails/groupReplays/useReplaysFromIssue.tsx [hunk 1]
```diff
@@ -51,15 +51,15 @@ function useReplayFromIssue({
...
+    if (!replayIds || !replayIds.length) {
       return null;
     }
     return EventView.fromSavedQuery({
       id: '',
       name: '',
       version: 2,
       fields: REPLAY_LIST_FIELDS,
-      query: `id:[${String(replayIds)}]`,
+      query: replayIds.length ? `id:[${String(replayIds)}]` : `id:1`,
```
Comment (ryan953) [inline static/app/views/issueDetails/groupReplays/useReplaysFromIssue.tsx:62]:
```suggestion
      [query: replayIds.length ? `id:[${String(replayIds)}]` : `id:1`,](query: `id:[${String(replayIds)}]`,)
```

File: static/app/utils/replays/fetchReplayList.tsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ async function fetchReplayList({
         // when queryReferrer === 'issueReplays' we override the global view check on the backend
         // we also require a project param otherwise we won't yield results
         queryReferrer,
-        project: queryReferrer === 'issueReplays' ? ALL_ACCESS_PROJECTS : payload.project,
```
Comment (ryan953) [inline static/app/utils/replays/fetchReplayList.tsx:62]: This wasn't doing anything before. We only call `fetchReplayList` from the hook `useReplayList`... and that hook is only used in 4 places, all are updated in this PR to have specific referrers now.

Also the types prevented any other values from appearing here.

Comment (ryan953) [reply static/app/utils/replays/fetchReplayList.tsx:62]: and... this might actually be doing something on the backend.

Comment (ryan953) [reply static/app/utils/replays/fetchReplayList.tsx:62]: Yup, right here: https://github.com/getsentry/sentry/blob/c5d004cb0f0c94547df196e29f4d9fb0b53daa14/src/sentry/replays/endpoints/organization_replay_index.py#L41-L44

**Pull request merged**